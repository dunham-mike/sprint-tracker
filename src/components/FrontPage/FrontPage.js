import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import { Link as RouterLink } from 'react-router-dom';

import Button from '@material-ui/core/Button';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import Container from '@material-ui/core/Container';
import Link from '@material-ui/core/Link';
import FavoriteIcon from '@material-ui/icons/Favorite';

function Copyright() {
  return (
    <Typography variant="body2" color="textSecondary" align="center">
      {'Copyright Â© '}
      <Link color="inherit" href="https://github.com/dunham-mike" target="_blank" rel="noopener noreferrer">
        Mike Dunham
      </Link>{' '}
      {new Date().getFullYear()}
      {'.'}
    </Typography>
  );
}

const styles = theme => ({
  icon: {
    marginRight: theme.spacing(2),
  },
  heroContent: {
    backgroundColor: theme.palette.background.paper,
    padding: theme.spacing(8, 0, 6),
  },
  heroButtons: {
    marginTop: theme.spacing(4),
  },
  buttonContainer: {
    justifyContent: "center",
    '@media (max-width: 500px)': {
        flexDirection: 'column',
    },
  },
  buttonGridItem: {
    '@media (max-width: 500px)': {
        margin: '0 auto',
    },
  },
  cardGrid: {
    paddingTop: theme.spacing(8),
    paddingBottom: theme.spacing(8),
  },
  cardContainer: {
    display: 'flex',
    justifyContent: 'center'
  },
  card: {
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
  },
  cardMedia: {
    paddingTop: '56.25%', // 16:9
  },
  cardContent: {
    flexGrow: 1,
  },
  tryDemo: {
      marginTop: theme.spacing(-2),
      marginBottom: theme.spacing(6),
      textAlign: 'center',
  },
  footer: {
    backgroundColor: theme.palette.background.paper,
    padding: theme.spacing(6),
  },
  copyright: {
    marginTop: theme.spacing(0.3)
  }
});

const cards = [
    {
        imageUrl: 'https://images.unsplash.com/photo-1568584263125-bf8f0a77d51c?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1051&q=80',
        heading: 'Simple, structured sprints.',
        content: 'Purposely designed for lightweight data collection, so individuals can be flexible in how they manage their work.',
    }, 
    {
        imageUrl: 'https://images.unsplash.com/photo-1559526323-cb2f2fe2591b?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1050&q=80',
        heading: 'Weekly progress tracking.',
        content: 'Updating and seeing a sprint\'s progress is a breeze.',
    }, 
    {
        imageUrl: 'https://images.unsplash.com/photo-1581291518857-4e27b48ff24e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1050&q=80',
        heading: 'Authentication built-in.',
        content: 'Make sure your work stays private.',
    }, 
];

const frontPage = (props) => {
    const { classes } = props;

    return(
        <React.Fragment>
            <main>
            {/* Hero unit */}
            <div className={classes.heroContent}>
            <Container maxWidth="sm">
                <Typography component="h1" variant="h2" align="center" color="textPrimary" gutterBottom>
                Deliberate Sprints
                </Typography>
                <Typography variant="h5" align="center" color="textSecondary" paragraph>
                Sprint tracking. With a point of view.
                </Typography>
                <div className={classes.heroButtons}>
                <Grid container spacing={2} className={classes.buttonContainer}>
                    <Grid item className={classes.buttonGridItem}>
                        <Button variant="contained" color="primary" component={RouterLink} to="/demo">
                            TRY A DEMO
                        </Button>
                    </Grid>
                    <Grid item className={classes.buttonGridItem}>
                        <Button variant="outlined" color="primary" component={RouterLink} to="/create-account">
                            CREATE AN ACCOUNT
                        </Button>
                    </Grid>
                    <Grid item className={classes.buttonGridItem}>
                        <Button variant="outlined" color="primary" component={RouterLink} to="/login">
                            LOG IN
                        </Button>
                    </Grid>
                </Grid>
                </div>
            </Container>
            </div>
            <Container className={classes.cardGrid} maxWidth="md">
            {/* End hero unit */}
            <Grid container spacing={4} className={classes.cardContainer}>
                {cards.map((card, index) => (
                    <Grid item key={card.heading} xs={11} sm={index === 0 ? 8 : 6} md={4}>
                        <Card className={classes.card}>
                        <CardMedia
                            className={classes.cardMedia}
                            image={card.imageUrl}
                            title={card.heading}
                        />
                        <CardContent className={classes.cardContent}>
                            <Typography gutterBottom variant="h5" component="h2">
                            {card.heading}
                            </Typography>
                            <Typography>
                            {card.content}
                            </Typography>
                        </CardContent>
                        </Card>
                    </Grid>
                ))}
            </Grid>
            </Container>
            <div className={classes.tryDemo}>
            <Container maxWidth="sm">
                <Button variant="outlined" color="secondary" component={RouterLink} to="/create-account">
                    Create your free account
                </Button>
            </Container>
            </div>
        </main>
        {/* Footer */}
        <footer className={classes.footer}>
            <Typography variant="h6" align="center" gutterBottom>
            Built with React and Material UI.
            </Typography>
            <Typography variant="subtitle1" align="center" color="textSecondary" component="p">
                Made with <FavoriteIcon fontSize="small" color={"secondary"} /> in Burlingame, CA.
            </Typography>
            <div className={classes.copyright}>
                <Copyright  />
            </div>
            
        </footer>
      </React.Fragment>
    );
};

export default withStyles(styles)(frontPage);