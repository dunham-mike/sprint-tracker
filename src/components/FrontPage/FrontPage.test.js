import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import FrontPage from './FrontPage';

describe('Render FrontPage', () => {

    it('renders FrontPage component correctly', () => {  
        const FrontPageComponent = renderer.create(<BrowserRouter><FrontPage /></BrowserRouter>).toJSON();
        expect(FrontPageComponent).toMatchSnapshot();
    });

});