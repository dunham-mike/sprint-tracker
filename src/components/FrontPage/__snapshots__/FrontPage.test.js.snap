// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render FrontPage renders FrontPage component correctly 1`] = `
Array [
  <main>
    <div
      className="hero-heroContent-8"
    >
      <div
        className="MuiContainer-root MuiContainer-maxWidthSm"
      >
        <h1
          className="MuiTypography-root MuiTypography-h2 MuiTypography-colorTextPrimary MuiTypography-gutterBottom MuiTypography-alignCenter"
        >
          Deliberate Sprints
        </h1>
        <p
          className="MuiTypography-root MuiTypography-h5 MuiTypography-colorTextSecondary MuiTypography-paragraph MuiTypography-alignCenter"
        >
          Sprint tracking. With a point of view.
        </p>
        <div
          className="hero-heroButtons-9"
        >
          <div
            className="MuiGrid-root hero-buttonContainer-10 MuiGrid-container MuiGrid-spacing-xs-2"
          >
            <div
              className="MuiGrid-root hero-buttonGridItem-11 MuiGrid-item"
            >
              <a
                aria-disabled={false}
                className="MuiButtonBase-root MuiButton-root MuiButton-contained hero-button-12 MuiButton-containedPrimary"
                href="/demo"
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                <span
                  className="MuiButton-label"
                >
                  TRY A DEMO
                </span>
              </a>
            </div>
            <div
              className="MuiGrid-root hero-buttonGridItem-11 MuiGrid-item"
            >
              <a
                aria-disabled={false}
                className="MuiButtonBase-root MuiButton-root MuiButton-outlined hero-button-12 MuiButton-outlinedPrimary"
                href="/create-account"
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                <span
                  className="MuiButton-label"
                >
                  CREATE AN ACCOUNT
                </span>
              </a>
            </div>
            <div
              className="MuiGrid-root hero-buttonGridItem-11 MuiGrid-item"
            >
              <a
                aria-disabled={false}
                className="MuiButtonBase-root MuiButton-root MuiButton-outlined hero-button-12 MuiButton-outlinedPrimary"
                href="/login"
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                <span
                  className="MuiButton-label"
                >
                  LOG IN
                </span>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiContainer-root frontPage-cardGrid-2 MuiContainer-maxWidthMd"
    >
      <div
        className="MuiGrid-root frontPage-cardContainer-3 MuiGrid-container MuiGrid-spacing-xs-4"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11 MuiGrid-grid-sm-8 MuiGrid-grid-md-4"
        >
          <div
            className="MuiPaper-root MuiCard-root frontPage-card-4 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              className="MuiCardMedia-root frontPage-cardMedia-5"
              style={
                Object {
                  "backgroundImage": "url(\\"https://images.unsplash.com/photo-1568584263125-bf8f0a77d51c?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1051&q=80\\")",
                }
              }
              title="Simple, structured sprints."
            />
            <div
              className="MuiCardContent-root frontPage-cardContent-6"
            >
              <h2
                className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
              >
                Simple, structured sprints.
              </h2>
              <p
                className="MuiTypography-root MuiTypography-body1"
              >
                Purposely designed for lightweight data collection, so individuals can be flexible in how they manage their work.
              </p>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
        >
          <div
            className="MuiPaper-root MuiCard-root frontPage-card-4 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              className="MuiCardMedia-root frontPage-cardMedia-5"
              style={
                Object {
                  "backgroundImage": "url(\\"https://images.unsplash.com/photo-1559526323-cb2f2fe2591b?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1050&q=80\\")",
                }
              }
              title="Weekly progress tracking."
            />
            <div
              className="MuiCardContent-root frontPage-cardContent-6"
            >
              <h2
                className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
              >
                Weekly progress tracking.
              </h2>
              <p
                className="MuiTypography-root MuiTypography-body1"
              >
                Updating and seeing a sprint's progress is a breeze.
              </p>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
        >
          <div
            className="MuiPaper-root MuiCard-root frontPage-card-4 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              className="MuiCardMedia-root frontPage-cardMedia-5"
              style={
                Object {
                  "backgroundImage": "url(\\"https://images.unsplash.com/photo-1581291518857-4e27b48ff24e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1050&q=80\\")",
                }
              }
              title="Authentication built-in."
            />
            <div
              className="MuiCardContent-root frontPage-cardContent-6"
            >
              <h2
                className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
              >
                Authentication built-in.
              </h2>
              <p
                className="MuiTypography-root MuiTypography-body1"
              >
                Make sure your work stays private.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="frontPage-tryDemo-7"
    >
      <div
        className="MuiContainer-root MuiContainer-maxWidthSm"
      >
        <a
          aria-disabled={false}
          className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
          href="/create-account"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            className="MuiButton-label"
          >
            Create your free account
          </span>
        </a>
      </div>
    </div>
  </main>,
  <footer
    className="footer-footer-219"
  >
    <h6
      className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom MuiTypography-alignCenter"
    >
      Built with React and Material UI.
    </h6>
    <p
      className="MuiTypography-root footer-flexDisplay-221 MuiTypography-subtitle1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
    >
      <span>
        Made with 
      </span>
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeSmall"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
        />
      </svg>
      <span>
         in Burlingame, CA.
      </span>
    </p>
    <p
      className="MuiTypography-root footer-copyright-220 MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
    >
      © 
      <a
        className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
        href="https://www.mikedunham.org"
        onBlur={[Function]}
        onFocus={[Function]}
        rel="noopener noreferrer"
        target="_blank"
      >
        Mike Dunham
      </a>
       
      2020
    </p>
  </footer>,
]
`;
